decidim-proposals/lib/decidim/proposals/component.rb

```diff
@@ -38,7 +37,9 @@
     settings.attribute :allow_card_image, type: :boolean, default: false
     settings.attribute :resources_permissions_enabled, type: :boolean, default: true
     settings.attribute :collaborative_drafts_enabled, type: :boolean, default: false
-    settings.attribute :participatory_texts_enabled, type: :boolean, default: false
+    settings.attribute :participatory_texts_enabled,
+                       type: :boolean, default: false,
+                       disabled: ->(context) { Decidim::Proposals::Proposal.where(component: context[:component]).any? }
```

Comment:
Does this mean that the admin will only be able to enable participatory texts if there are some proposals for the current component?
We must take into account that participatory texts should be enabled for the admins to upload the text that will be converted into proposals.
Am I correct?

Reply:
Mmm, it's just the other way around: this setting will only be allowed to be changed when there are no proposals in the component. 😄

Reply:
Oh, I think that now I understand your concern. 

Maybe is a naming issue, if I rename from `disabled` to `readonly` it would be clearer for you?

I mean, the idea is that admins can change the value for this setting until the component has any proposal. After that, the setting's value just keeps the same, doesn't change to `false`.